### PR TITLE
feat: Fetch keys from back-end

### PR DESCRIPTION
**Pull request by stanleyjones**

### CODE REVIEW

File: src/features/services/queries.ts [hunk 1]
```diff
@@ -114,3 +115,24 @@ export function usePutValue() {
     },
   });
 }
+
+export function useGetValue() {
+  // eslint-disable-next-line
+  const [_, network] = useNetworkContext();
+  const keys = useDataServiceStore((s) => s.keys);
```
Comment (stanleyjones) [inline src/features/services/queries.ts:122]: Get the keys from state here...

File: src/features/services/queries.ts [hunk 2]
```diff
@@ -114,3 +115,24 @@ export function usePutValue() {
...
+
+  const queries = useQueries<KVStoreValue[]>({
+    queries: keys.map((key) => ({
```
Comment (stanleyjones) [inline src/features/services/queries.ts:125]: Fire off multiple queries, one for each key...

File: src/features/services/queries.ts [hunk 3]
```diff
@@ -114,3 +115,24 @@ export function usePutValue() {
...
+      queryKey: ["kvstore.get", key, network?.url],
+      queryFn: async () => {
+        const { value } = await network?.kvStore.get({ key });
```
Comment (stanleyjones) [inline src/features/services/queries.ts:128]: For each one, call `kvStore.get` and send `{ key: key }` as param...

File: src/features/services/queries.ts [hunk 4]
```diff
@@ -114,3 +115,24 @@ export function usePutValue() {
...
+        return { key, value: value.toString() };
```
Comment (stanleyjones) [inline src/features/services/queries.ts:129]: Make sure to add `key` back into the response, which isn't included...

File: src/features/services/queries.ts [hunk 5]
```diff
@@ -114,3 +115,24 @@ export function usePutValue() {
...
+      },
+    })),
+  });
+  return {
+    isLoading: queries.some((query) => query.isLoading),
+    isError: queries.some((query) => query.isError),
+    data: queries.map(({ data }) => data),
```
Comment (stanleyjones) [inline src/features/services/queries.ts:136]: ...and finally map them back into an array.

**Review by stanleyjones — COMMENTED** [review]
Here ya go.

**Review by mjsorribas — APPROVED** [review]

**Comment by stanleyjones** [comment]
I'll go ahead and merge this into my `store-keys` branch so you can use it, @mjsorribas .

**Pull request merged**